src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamespaceSymbol.cs

```diff
@@ -51,13 +51,22 @@ internal sealed override NamespaceExtent Extent
             }
         }
 
+        [PerformanceSensitive(
+            "https://github.com/dotnet/roslyn/issues/23582",
+            Constraint = "Provide " + nameof(ArrayBuilder<Symbol>) + " capacity to reduce number of allocations.")]
         public sealed override ImmutableArray<Symbol> GetMembers()
         {
             EnsureAllMembersLoaded();
 
-            var builder = ArrayBuilder<Symbol>.GetInstance();
-            builder.AddRange(GetMemberTypesPrivate());
-            builder.AddRange(GetMemberNamespacesPrivate());
+            var memberTypes = GetMemberTypesPrivate();
+            var builder = ArrayBuilder<Symbol>.GetInstance(memberTypes.Length + lazyNamespaces.Count);
+
+            builder.AddRange(memberTypes);
+            foreach (var pair in lazyNamespaces)
+            {
+                builder.Add(pair.Value);
+            }
```

Comment:
```builder.AddRange(lazyNamespaces.Values);```?

Reply:
Or foreach over ```lazyNamespaces.Values```

---
In reply to: [155110989](https://github.com/dotnet/roslyn/pull/23552#discussion_r155110989) [](ancestors = 155110989)

Reply:
Note: .Values allocates an object, so enumerating over the dictionary itself has been generally preferred.  We could add an overload to the builder . Something like "AddDictionaryValues to help out here so we don't have to have so many foreaches all over the place.

Reply:
I added the `AllowGenericEnumeration = false` constraint which, combined with dotnet/roslyn-analyzers#1439, should prevent regressions here.